torch/csrc/jit/codegen/cuda/lower_predicate.cpp

```diff
@@ -42,6 +42,33 @@ class ConditionalFromPredicateModifier : public kir::IrVisitor {
     if (expr != nullptr && expr->predicate() != nullptr) {
       // Replace expr predicate with bool conditional
       auto conditional = generateConditional(expr->predicate());
+      if (expr->predicate()->predicate_type() == PredicateType::Vectorize) {
+        // TODO: This logic doesn't seem to fit well here, for unswitch the
```

Comment:
Maybe `UnrollPass::handle(Expr*)`, where we first create `Vectorize` predicates?